### DISCUSSION TITLE
Wald ratio with delta method

### DISCUSSION THREAD

**Discussion by LucilleAugusta** [MRCIEU/TwoSampleMR#476] [Q&A] [answered]
Hi! 

I would like to use Wald ratio and the delta method. 
When looking in the code I find both `mr_meta_fixed` and `mr_meta_random` muted from the` mr_method_list `- is there a reason for this? [[Link to code](https://github.com/MRCIEU/TwoSampleMR/blob/f247a68d6ff32733e3c0a33c60d987340dc9c44c/R/mr.R#L365)]

I would like to know if there is anything wrong with the calculations or a better way to do it. 

I hope some one has some input to this. 
Thanks, 
Lucille

**Comment by mightyphil2000** [comment, marked as answer]
Hi Lucille

The quick answer is that those functions have been replaced with mr_ivw_fe (equivalent to mr_meta_fixed ) and mr_ivw_mre (better version of mr_meta_random). These methods are appropriate when you have multiple independent SNPs available to instrument your exposure. 

Longer answer:
The mr_meta_fixed method is equivalent to mr_ivw_fe. They should give the same result. mr_meta_fixed works by taking a weighted average of Wald ratios (using inverse variance of the SNP-outcome effect as the weight). mr_ivw_fe is the slope from a regression model of the SNP-outcome effects regressed on the SNP-exposure effects, with the standard error scaled to reflect a fixed effects model (this is done by removing a correction for over-dispersion, which is implemented by default in the regression model). 

mr_meta_random works similarly to mr_meta_fixed, except that the weights additionally incorporate information about heterogeneity between the Wald ratios (the fixed effects method assumes there is no heterogeneity). We have replaced mr_meta_random with mr_ivw_mre (**m**ultiplicative **r**andom **e**ffects). The mr_meta_random method implements an additive random effects model, whereas mr_ivw_mre is a multiplicative random effects model, which we think is a better method (for various reasons that we can discuss further if interested). 

bw
Philip

**Reply by LucilleAugusta** [reply]
Thanks for the quick response! 
I'll look into these functions.

/Lucille
